File: Programming Basics with Python/For-цикъл/oscars.py

actor = input()
points = float(input())
judges = int(input())

total_points = points
for i in range(1, judges+1):
    judge_name = input()
    judge_points = float(input())

    current_points = len(judge_name)* judge_points / 2
    total_points += current_points

    if total_points >= 1250.5:
        break

if total_points >= 1250.5:
    print(f"Congratulations, {actor} got a nominee for leading role with {total_points:.1f}!")
else:
    diff_points = 1250.5 - total_points
    print(f"Sorry, {actor} you need {diff_points:.1f} more!")

File: Programming Basics with Java/For-loop/Oscars.java
import java.util.Scanner;

public class Oscars 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        String actor = input.nextLine();
        double points = Double.parseDouble(input.nextLine());
        int n = Integer.parseInt(input.nextLine());

        for (int i = 0; i < n; i++)
        {
            if (points > 1250.5)
            {
                System.out.printf("Congratulations, %s got a nominee for leading role with %.1f!", actor, points);
                break;
            }
            String judge = input.nextLine();
            double judge_points = Double.parseDouble(input.nextLine());

            points += (judge.length() * judge_points) / 2;
        }

        if (points < 1250.5)
        {
            System.out.printf("Sorry, %s you need %.1f more!", actor, Math.abs(1250.5 - points));
        }
    }
}
